docs/deploy/configurations.md

```diff
@@ -83,7 +83,7 @@ Currently, you can set only 1 namespace to watch in this flag. See [this Kuberne
 |enable-waf                             | boolean                         | true            | Enable WAF addon for ALB |
 |enable-wafv2                           | boolean                         | true            | Enable WAF V2 addon for ALB |
 |external-managed-tags                  | stringList                      |                 | AWS Tag keys that will be managed externally. Specified Tags are ignored during reconciliation |
-|feature-gate                         | string                          | true             | A set of key=value pairs to enable or disable features |
+|feature-gate                           | string                          | boolean         | A set of key=value pairs to enable or disable features |
```

Comment:
the type of "feature-gate" should be stringMap instead of string.
the default value of "feature-gate" can be left empty here. and we provide detailed explain for each feature in a separate section like the "### disable-ingress-class-annotation"

Reply:
Also provide a link to the feature gates section for further reference

Reply:
recommend modifying the flag to `feature-gates` instead since there are more than one feature

Reply:
hmmm, if we use plural, should we use `features-gate` instead?

Reply:
nvm, will use `feature-gates`.